### PR TITLE
Remove `doc(include)`

### PR DESCRIPTION
**Pull request by jyn514**
This nightly feature is redundant now that `extended_key_value_attributes` is stable (https://github.com/rust-lang/rust/pull/83366). @rust-lang/rustdoc not sure if you think this needs FCP; there was already an FCP in #82539, but technically it was for deprecating, not removing the feature altogether.

This should not be merged before #83366.

cc @petrochenkov

### CODE REVIEW

File: src/tools/clippy/clippy_lints/src/missing_doc.rs [hunk 1]
```diff
@@ -94,7 +79,7 @@ impl MissingDoc {
         }
 
         let has_doc = attrs.iter().any(|a| {
-            a.is_doc_comment() || a.doc_str().is_some() || a.value_str().is_some() || Self::has_include(a.meta())
+            a.is_doc_comment() || a.doc_str().is_some() || a.value_str().is_some()
         });
```
Comment (jyn514) [inline src/tools/clippy/clippy_lints/src/missing_doc.rs:83]: FYI @rust-lang/clippy you should probably be using `rustc_lint::builtin::has_doc` instead; this has a false positive and also counts things like `#[path = "x"]` as doc-comments (because you only check value_str().is_some(), not the name of the attribute). Also in general it seems nice to have all the logic in one place.

Comment (phansch) [reply src/tools/clippy/clippy_lints/src/missing_doc.rs:83]: Looks like a nice cleanup opportunity on the Clippy side! `has_doc` would just have to be made public

Comment (jyn514) [reply src/tools/clippy/clippy_lints/src/missing_doc.rs:83]: Sure thing, you could do both at the same time in a PR to rust-lang/rust :)

**Comment by GuillaumeGomez** [comment]
> This nightly feature is redundant now that `extended_key_value_attributes` is stable (#83366). @rust-lang/rustdoc not sure if you think this needs FCP; there was already an FCP in #82539, but technically it was for deprecating, not removing the feature altogether.

This is a nightly feature, so I'd say that the FCP is unneeded here.

File: src/tools/clippy/clippy_lints/src/missing_doc.rs (hunk 1 above)
Comment (camsteffen) [reply src/tools/clippy/clippy_lints/src/missing_doc.rs:83]: I opened an issue: rust-lang/rust-clippy#7247

**Comment by jyn514** [comment]
@GuillaumeGomez do you mind reviewing the rustdoc changes? They should be pretty simple since it's just deleted code.

Oh, I need to add a diagnostic test for suggesting `doc = include_str!`, let me write that up first.

**Comment by GuillaumeGomez** [comment]
> Oh, I need to add a diagnostic test for suggesting `doc = include_str!`, let me write that up first.

Not sure this is needed though. But as you prefer. Ping me once you want me to review.

**Comment by jyn514** [comment]
@GuillaumeGomez ok, this is ready for review

File: compiler/rustc_feature/src/removed.rs [hunk 2]
```diff
@@ -140,6 +140,9 @@ declare_features! (
     (removed, const_fn, "1.54.0", Some(57563), None,
      Some("split into finer-grained feature gates")),
 
+    /// Allows `#[doc(include = "some-file")]`.
+    (removed, external_doc, "1.54.0", Some(44732), None, Some("use #[doc = include_str!(\"filename\")] instead, which handles macro invocations")),
```
Comment (petrochenkov) [inline compiler/rustc_feature/src/removed.rs:144]: Nit: it still makes sense to keep the 100 character line limit even if rustfmt doesn't enforce it in macros.

File: compiler/rustc_passes/src/check_attr.rs [hunk 3]
```diff
@@ -705,7 +705,7 @@ impl CheckAttrVisitor<'tcx> {
         let mut is_valid = true;
 
         if let Some(list) = attr.meta().and_then(|mi| mi.meta_item_list().map(|l| l.to_vec())) {
-            for meta in list {
+            for meta in &list {
```
Comment (GuillaumeGomez) [inline compiler/rustc_passes/src/check_attr.rs:708]: What is the reason behind this change?

File: src/test/ui/rustdoc/doc-include-suggestion.fixed [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// run-rustfix
```
Comment (GuillaumeGomez) [inline src/test/ui/rustdoc/doc-include-suggestion.fixed:1]: The extension of this file is `.fixed`? O.o I didn't know that one.

File: compiler/rustc_passes/src/check_attr.rs (hunk 3 above)
Comment (jyn514) [reply compiler/rustc_passes/src/check_attr.rs:708]: It avoids a borrow error below, where I call `list.len()`.

File: src/test/rustdoc/external-doc.rs [hunk 5]
```diff
@@ -1,12 +1,3 @@
-#![feature(external_doc)]
-
-// @has external_doc/struct.CanHasDocs.html
-// @has - '//h1' 'External Docs'
-// @has - '//h2' 'Inline Docs'
-#[doc(include = "auxiliary/external-doc.md")]
```
[the remark this answers is not in the document]
Comment (jyn514) [reply src/test/rustdoc/external-doc.rs:6]: That's already tested immediately below: https://github.com/rust-lang/rust/blob/70a79ccb50970545613f9c8bd1c7b3821f1b61da/src/test/rustdoc/external-doc.rs#L4

**Comment by jyn514** [comment]
@GuillaumeGomez this is ready for a second round of review

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks for this! This was a long awaited removal. :)

@bors: r+
